### PR TITLE
Feature/no transfer unread

### PR DESCRIPTION
**Pull request by glassresistor**

### CODE REVIEW

**Comment by glassresistor** [comment]
@tmaybe please review and merge

File: spec/features/admin_spec.rb [hunk 1]
```diff
@@ -325,5 +325,60 @@
...
+          fill_in 'transfer_note', with: 'Notes notes notes.'
+
+          click_on 'Transfer Client'
+
+          expect(page).to have_content('Transfer Client and Mark Messages As Read')
+          expect(page).to have_content('Change user')
+          expect(page).to have_select("user_in_dept_#{department1.id}", options: ['', user1.full_name, user2.full_name])
+          expect(page).to have_content('Include a message for the new user')
+
+          expect(page).to have_content('This client has unread messages. The messages will not be transferred to the new user.')
```
Comment (tmaybe) [inline spec/features/admin_spec.rb:365]: use the I18n string `I18n.t('views.admin.reporting_relationships.edit.unread_transfer')`

File: spec/features/admin_spec.rb [hunk 2]
```diff
@@ -325,5 +325,60 @@
...
+          click_on('Change')
+
+          expect(page).to have_content('Transfer Client')
+          expect(page).to have_content('Change user')
+          expect(page).to have_select("user_in_dept_#{department1.id}", options: ['', user1.full_name, user2.full_name])
+          expect(page).to have_content('Include a message for the new user')
+        end
+
+        step 'a new message has come in and an error is shown' do
+          msg = create :message, reporting_relationship: ReportingRelationship.find_by(user: user1, client: client1), read: false
```
Comment (tmaybe) [inline spec/features/admin_spec.rb]: Set `has_unread_messages` on the reporting relationship to true as well.

File: spec/features/admin_spec.rb [hunk 3]
```diff
@@ -325,5 +325,60 @@
...
+          expect(page).to have_content('Include a message for the new user')
+
+          expect(page).to have_content('This client has unread messages. The messages will not be transferred to the new user.')
+
+          select user2.full_name, from: 'Transfer to'
+          fill_in 'transfer_note', with: 'Notes notes notes.'
+
+          click_on 'Transfer Client and Mark Messages As Read'
+
+          expect(msg.reload).to be_read
```
Comment (tmaybe) [inline spec/features/admin_spec.rb:372]: expect `has_unread_messages` on the reporting relationship to be false as well.

File: spec/features/user_edits_client_spec.rb [hunk 4]
```diff
@@ -90,6 +91,50 @@
     end
   end
 
+  context 'there are unread messages' do
+    let(:rr) { ReportingRelationship.find_by(user: my_user, client: clientone) }
+
+    before do
+      create :message, reporting_relationship: rr, read: false, inbound: true
```
Comment (tmaybe) [inline spec/features/user_edits_client_spec.rb:98]: set `has_unread_messages` on the reporting relationship to true as well

File: spec/features/user_edits_client_spec.rb [hunk 5]
```diff
@@ -22,6 +22,7 @@
   let(:new_phone_number) { '2024042234' }
   let(:new_phone_number_display) { '(202) 404-2234' }
 
+  let(:unread_error_message) { 'This client has unread messages. The messages will not be transferred to the new user. Transfer now, or click here to read them.' }
```
Comment (tmaybe) [inline spec/features/user_edits_client_spec.rb:25]: use the I18n string `I18n.t('views.clients.edit.manage.unread_transfer')`

File: spec/features/user_edits_client_spec.rb [hunk 6]
```diff
@@ -90,6 +91,50 @@
...
+  context 'a message is received while editing' do
+    let(:rr) { ReportingRelationship.find_by(user: my_user, client: clientone) }
+
+    scenario 'the page displays the warning when the message is received' do
+      within "#client_#{clientone.id}" do
+        find('td', text: 'Manage').click
+      end
+
+      expect(page).to have_current_path(edit_client_path(clientone))
+      expect(page).to_not have_content('This client has unread messages. The messages will not be transferred to the new user. Transfer now, or click here to read them.')
```
Comment (tmaybe) [inline spec/features/user_edits_client_spec.rb]: Use the variable for this string (and/or the I18n value)

File: spec/features/user_edits_client_spec.rb [hunk 7]
```diff
@@ -90,6 +91,50 @@
...
+      expect(page).to have_content(unread_error_message)
+
+      within '#unread-warning' do
+        click_on 'click here'
+      end
+
+      expect(page).to have_current_path(reporting_relationship_path(rr))
+    end
+  end
+
```
Comment (tmaybe) [inline spec/features/user_edits_client_spec.rb:141]: Add a test for what happens when there's a warning and the user transfers the client; it should check the `read` value of the message and the `has_unread_messages` value of the reporting relationship.

**Review by tmaybe — CHANGES_REQUESTED** [review]
I added some comments for suggested changes to tests.

Also, verify with @alsomanya that this current behavior is correct: we don't mark messages read if it's a user doing the transfer.

**Comment by glassresistor** [comment]
@alsomanya says to mark them as read.  so the code should work as implemented.  working on i18n stuff now.  Wanted to but i think we might have to transform some.

I'll add the has_unread_message checks and make sure that happens, good catch,

**Pull request closed without merging**